### PR TITLE
Implement ConfigurationService server for local dev support

### PR DESCRIPTION
**Pull request by DavidSGK**
# Details
- Updated SDKServer class to also implement ConfigurationService methods
    - Also handle reading specific headers from requests to allow reinitializing for different repo paths
- Updated remote-only client initialization to be non-async
- Added binary script `lekko-server` that starts a config server that can be connected to by other SDKs (tested with js-sdk)

Example usage
```
# In a project using this SDK...
npm install --save @lekko/node-server-sdk
# You can invoke server script via npx
npx lekko-server
# Or yarn if you use yarn
yarn lekko-server
# Can also pass some args
npx lekko-server --port 50051 --repo-path ~/path/to/repo
```

### CODE REVIEW

File: src/memory/server.ts [hunk 1]
```diff
@@ -1,34 +1,162 @@
...
+                        throw e;
+                    }
                 }
             });
-        this.server = createServer(connectNodeAdapter({ routes })).listen(port);
+        };
+        this.server = http.createServer(connectNodeAdapter({ routes })).listen(port);
+    }
+
+    fromReqContext(context: { [key: string]: Value }): ClientContext {
```
Comment (passichenko) [inline src/memory/server.ts:117]: Maybe make it a static method in `ClientContext`? Similar to `fromJSON`.

File: src/server.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
```
Comment (passichenko) [inline src/server.ts:1]: nit: is it really needed?

File: src/transport-builder.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export class ClientTransportBuilder {
     this.apiKey = apiKey;
   }
 
-  async build(): Promise<Transport> {
```
Comment (passichenko) [inline src/transport-builder.ts:28]: lol, so it wasn't actually async?

File: src/memory/server.ts [hunk 4]
```diff
@@ -1,34 +1,162 @@
...
+                listContents: async () => {
+                    return this.client.listContents();
+                }
+            });
+            // TODO: Clean up code once interceptor support lands in connect-node
+            router.service(ConfigurationService, {
+                getBoolValue: async (req, context) => {
+                    await this.handleHeaders(context.requestHeader);
+                    try {
+                        await this.handleHeaders(context.requestHeader);
```
Comment (passichenko) [inline src/memory/server.ts:36]: unless you really need to do it twice :)
```suggestion
                    await this.handleHeaders(context.requestHeader);
                    try {
```

File: src/memory/server.ts [hunk 5]
```diff
@@ -1,34 +1,162 @@
...
+
+    logEval(type: string, namespace: string, key: string): void {
+        // eslint-disable-next-line no-console
+        console.log(`Evaluated ${type} config ${namespace}/${key}`);
+    }
+
+    async handleHeaders(headers: Headers): Promise<void> {
+        const localPath = headers.get(LOCAL_PATH_HEADER);
+        if (localPath) {
+            await this.client.reinitialize({ path: localPath });
```
Comment (passichenko) [inline src/memory/server.ts:150]: why is this needed?

**Comment by passichenko** [comment]
> Also handle reading specific headers from requests to allow reinitializing for different repo paths

I'm not following. Aren't we set repo path on startup?

File: src/memory/server.ts (hunk 4 above)
Comment (DavidSGK) [reply src/memory/server.ts]: Oops, nice catch!

File: src/transport-builder.ts (hunk 3 above)
Comment (DavidSGK) [reply src/transport-builder.ts:28]: Yeah building the transport object itself isn't async so I just cleaned it up here - but unfortunately the client initializations still need to be :/

File: src/server.ts (hunk 2 above)
Comment (DavidSGK) [reply src/server.ts:1]: It actually is haha, for executable scripts to be handled correctly. This file is the entrypoint that gets built into `bin/server.js` which is referenced in `package.json`

File: src/memory/server.ts (hunk 1 above)
Comment (DavidSGK) [reply src/memory/server.ts]: I thought about doing that originally but decided against it because I didn't want this being a publicly visible method for end users, it only makes sense in the context (no pun intended) of converting from the ConfigurationService context proto to a client context

File: src/memory/server.ts (hunk 5 above)
Comment (DavidSGK) [reply src/memory/server.ts]: This is related to your other question below, so will answer there

**Comment by DavidSGK** [comment]
> > Also handle reading specific headers from requests to allow reinitializing for different repo paths
> 
> I'm not following. Aren't we set repo path on startup?

Yes, the repo path can be set on startup. But for end-user simplicity I wanted to give them the option to start the config server without any args (e.g. just `npx lekko-server`) and allow specifying the path in the react SDK instead. This also allows changing the repo path without having to restart the config server. That's what the `reinitialize` method logic is for.

File: src/memory/server.ts [hunk 6]
```diff
@@ -1,36 +1,213 @@
...
+      )
+      .listen(port);
+  }
+
+  fromReqContext(context: { [key: string]: Value }): ClientContext {
+    const clientContext = new ClientContext();
+    Object.entries(context).forEach(([key, value]) => {
+      switch (value.kind.case) {
+        case "boolValue": {
+          clientContext.setBoolean(key, value.kind.value);
```
Comment (passichenko) [inline src/memory/server.ts:171]: Looking at this again, I realized that `ClientContext` actually stores context values as `Value`, so here you unwrap it and then `setBoolean` will wrap it again. So looks like here you just need to make `ClientContext`'s constructor to accept `data` and pass `context` as is.

**Review by passichenko — APPROVED** [review]

File: src/memory/server.ts (hunk 5 above)
Comment (DavidSGK) [reply src/memory/server.ts]: https://github.com/lekkodev/js-sdk/pull/5 is where the headers would be set

File: src/memory/server.ts (hunk 6 above)
Comment (DavidSGK) [reply src/memory/server.ts:171]: Good call, will do!
